### PR TITLE
impl of frame callback and some cleanup and a bugfix

### PR DESCRIPTION
**Pull request by brianPlummer**

### CODE REVIEW

File: library/src/main/java/com/codemonkeylabs/fpslibrary/FPSFrameCallback.java [hunk 1]
```diff
@@ -41,27 +41,54 @@ public void doFrame(long frameTimeNanos)
...
         Choreographer.getInstance().postFrameCallback(this);
     }
 
+    private void collectSampleAndSend(long frameTimeNanos)
+    {
+        //this occurs only when we have gathered over the sample time ~700ms
+        List<Long> dataSetCopy = new ArrayList<>();
+        dataSetCopy.addAll(dataSet);
+
+        //push data to the controller
```
Comment (digitalbuddha) [inline library/src/main/java/com/codemonkeylabs/fpslibrary/FPSFrameCallback.java]: push data to the presenter

File: library/src/main/java/com/codemonkeylabs/fpslibrary/FPSFrameCallback.java [hunk 2]
```diff
@@ -41,27 +41,54 @@ public void doFrame(long frameTimeNanos)
...
+
+        // clear data
+        dataSet.clear();
+
+        //reset sample timer to last frame
+        startSampleTimeInNs = frameTimeNanos;
+    }
+
+    /**
+     * returns true if we have exceeded our sample length
```
Comment (digitalbuddha) [inline library/src/main/java/com/codemonkeylabs/fpslibrary/FPSFrameCallback.java]: returns true when sample length is exceed

File: library/src/main/java/com/codemonkeylabs/fpslibrary/FrameDataCallback.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.codemonkeylabs.fpslibrary;
+
+/**
+ * Created by brianplummer on 11/12/15.
+ */
+public interface FrameDataCallback
+{
+    /**
+     * this is called for every doFrame() on the choreographer callback
+     * use this very judiciously.  you don't even want to Log.d() from here, bad idea
```
Comment (digitalbuddha) [inline library/src/main/java/com/codemonkeylabs/fpslibrary/FrameDataCallback.java]: Logging synchronously from here is bad idea as doFrame will be called every 16-32ms.

File: library/src/main/java/com/codemonkeylabs/fpslibrary/ui/MeterPresenter.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package com.codemonkeylabs.fpslibrary.service;
+package com.codemonkeylabs.fpslibrary.ui;
```
Comment (digitalbuddha) [inline library/src/main/java/com/codemonkeylabs/fpslibrary/ui/MeterPresenter.java]: I like TinyCoach as a presenter name for TinyDancer. thoughts?

File: library/src/main/java/com/codemonkeylabs/fpslibrary/ui/MeterTouchListener.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package com.codemonkeylabs.fpslibrary.service;
+package com.codemonkeylabs.fpslibrary.ui;
```
Comment (digitalbuddha) [inline library/src/main/java/com/codemonkeylabs/fpslibrary/ui/MeterTouchListener.java]: DancerTouchListener?

**Comment by digitalbuddha** [comment]
lgtm, some comment copy edit.  I want to float the idea around of calling the widget/view a dancer. We can then build other tools/metrics that also use a DancerView.

**Comment by brianPlummer** [comment]
@digitalbuddha all done...gracias!  i'm going to push this as 0.0.3. and update readme once mavenCentral picks it up

**Pull request merged**